src/sql/parts/profiler/service/profilerService.ts

```diff
@@ -48,35 +49,36 @@ export class ProfilerService implements IProfilerService {
 	private _providers = new Map<string, sqlops.ProfilerProvider>();
 	private _idMap = new TwoWayMap<ProfilerSessionID, string>();
 	private _sessionMap = new Map<ProfilerSessionID, IProfilerSession>();
-	private _dialog: ProfilerColumnEditorDialog;
+	private _editColumnDialog: ProfilerColumnEditorDialog;
 
 	constructor(
 		@IConnectionManagementService private _connectionService: IConnectionManagementService,
 		@IConfigurationService public _configurationService: IConfigurationService,
 		@IInstantiationService private _instantiationService: IInstantiationService,
-		@INotificationService private _notificationService: INotificationService
+		@INotificationService private _notificationService: INotificationService,
+		@ICommandService private _commandService: ICommandService
 	) { }
 
 	public registerProvider(providerId: string, provider: sqlops.ProfilerProvider): void {
 		this._providers.set(providerId, provider);
 	}
 
-	public registerSession(uri: string, connectionProfile: IConnectionProfile, session: IProfilerSession): ProfilerSessionID {
+	public async registerSession(uri: string, connectionProfile: IConnectionProfile, session: IProfilerSession): Promise<ProfilerSessionID> {
 		let options: IConnectionCompletionOptions = {
 			params: { connectionType: ConnectionType.default, runQueryOnCompletion: RunQueryOnConnectionMode.none, input: undefined },
 			saveTheConnection: false,
 			showDashboard: false,
 			showConnectionDialogOnError: false,
 			showFirewallRuleOnError: true
 		};
-		this._connectionService.connect(connectionProfile, uri, options).then(() => {
+		await this._connectionService.connect(connectionProfile, uri, options).then(() => {
```

Comment:
Usually there's no need for `.then` and `.catch` when you're using async. It looks like this is just catching the connection error and ignoring it if it happens? I think a regular try/catch should work now since it's async